packages/client/lib/execution/vmexecution.ts

```diff
@@ -268,12 +268,25 @@ export class VMExecution extends Execution {
               if (!this.started) {
                 throw Error('Execution stopped')
               }
+              const beforeTS = Date.now()
               const result = await this.vm.runBlock({
                 block,
                 root: parentState,
                 skipBlockValidation,
                 skipHeaderValidation: true,
               })
+              const afterTS = Date.now()
+              const diffSec = Math.round((afterTS - beforeTS) / 1000)
+
+              if (diffSec > this.MAX_TOLERATED_BLOCK_TIME) {
```

Comment:
Note: this technically will only run if the block runs for at least 12.5 seconds

Reply:
Yeah, ok, I guess that's toerable. This is not "exact science" here anyhow. 🙂